File: unix/boot/spp/rpp/ratlibr/docant.r
include	defs

# docant
#
# Similar to cant(name), however precede the messge with the name
# of the program that was running when the file could not be
# opened.  Helpful in a pipeline to verify which program was not
# able to open a file.
#
   subroutine  docant(name)

   character  name(ARB), prog(FILENAMESIZE)
   integer    length
   integer    getarg

   length = getarg(0, prog, FILENAMESIZE)
   if (length != EOF) {
      call putlin(prog, STDERR)
      call putch(COLON, STDERR)
      call putch(BLANK, STDERR)
   }
   call cant(name)

   return
   end

File: unix/boot/spp/rpp/ratlibf/docant.f
      subroutine docant(name)
      integer name(100), prog(30)
      integer length
      integer getarg
      length = getarg(0, prog, 30)
      if (.not.(length .ne. -1))goto 23000
      call putlin(prog, 2)
      call putch(58, 2)
      call putch(32, 2)
23000 continue
      call cant(name)
      return
      end
